### PR TITLE
Add yellowNotesApi axios instance

### LINKED ISSUE #20: Implement API communication
**Guidelines**

- Use axios module
- Implement communication with these endpoints: 
- POST request to `/users/register` (register page) attached to 'Login' button
- POST (due to JWT authentication) request to `/users/authenticate`  (login page) attached to 'Create new account' button

### PR DESCRIPTION
**Pull request by filip-wan**
Fixes #20

### CODE REVIEW

File: src/contexts/UserContext/actions/addNoteAction.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  let response;
+  try {
+    // TODO API - remove and uncomment
+    response = {
+      status: 200,
+      data: { content: action.payload.content }
+    };
+    //response = await yellowNotesApi.post('user/note', { ...action.payload });
+  } catch (e) {
+    throw new Error('Registration request has did not succeed! ', response);
```
Comment (c1rcle) [inline src/contexts/UserContext/actions/addNoteAction.js]: Grammar error.

File: src/contexts/UserContext/actions/loginAction.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    {
+      id: 4,
+      content: 'Multi \nline \ntext'
+    }
+  ];
+
+  const { payload } = action;
+  const { email, password } = payload;
+
+  if (!payload || !password || !email || Object.keys(action.payload).length !== 2)
```
Comment (c1rcle) [inline src/contexts/UserContext/actions/loginAction.js]: Could you do it in a different way? 😃

File: src/contexts/UserContext/reducer.js [hunk 3]
```diff
@@ -1,12 +1,30 @@
 export default (state = { isUserLoggedIn: false }, { type, payload }) => {
   switch (type) {
+    case 'CHECK_TOKEN':
+      return { isUserLoggedIn: false, ...payload };
     case 'REGISTER':
       return { ...payload, isUserLoggedIn: true };
+    case 'REGISTER_FAILED':
+      // TODO: Real output
+      alert('That email is already occupied, you imposter!');
```
Comment (c1rcle) [inline src/contexts/UserContext/reducer.js]: These are also constant. They shouldn't be hard coded.

**Review by c1rcle — CHANGES_REQUESTED** [review]
Please, remove any comments that remain.

File: src/contexts/UserContext/reducer.js (hunk 3 above)
Comment (filip-wan) [reply src/contexts/UserContext/reducer.js]: It's just a placeholder for a real error handling.

File: src/contexts/UserContext/actions/addNoteAction.js (hunk 1 above)
Comment (filip-wan) [reply src/contexts/UserContext/actions/addNoteAction.js]: Thanks, I missed it.

**Review by c1rcle — APPROVED** [review]

**Pull request merged**